rust-lang/cargo
# bump jobserver to respect `--jobserver-auth=fifo:PATH`

### What does this PR try to resolve?

**tl;dr**: GNU Make 4.4 switches to use named pipes instead of anonymous simple pipes. Cargo needs to respect that to prevent clients from breaking.

See https://github.com/alexcrichton/jobserver-rs/pull/49

### How should we test and review this PR?

1. Get a GNU Make 4.4 binary.
2. Change lines like [this](https://github.com/rust-lang/cargo/blob/524231f43f602a553216548a81e4ae22be8dd905/tests/testsuite/jobserver.rs#L66) pointing to your binary. 
    For example, on macOS I changed to `gmake` to invoke GNU Make from homebrew.
3. Run `cargo t --test testsuite jobserver::`

If you want to test this patch live, you can also execute `cargo build` from `make -j2` to test the behaviour.
(I usually run [`procs -w rustc`](https://github.com/dalance/procs) to observe the amount of rustc invocations.)

```makefile
# Makefile
all:
	+./target/debug/cargo build --release
clean:
	cargo clean --release
```

### Additional information

Fallback workaround: add `--jobserver-style=pipe` if you are using GNU Make 4.4 and this PR has yet got landed.

Some distros that already got GNU Make 4.4:

- Fedora 38: https://src.fedoraproject.org/rpms/make/c/715e733cee430436b250af7d1e32d72fb7dfc8b6?branch=f38
- openSUSE Tumbleweed: https://build.opensuse.org/request/show/1032805
- Arch Linux, Manjaro: https://github.com/archlinux/svntogit-packages/commit/9448ab762d966b18025ec55f0a4239cbe6de8b0f
- Alpine Linux (edge): https://git.alpinelinux.org/aports/commit/main?id=9852cb19b76057ec5216f238ffbf5ff8d5de010f
- KaOS: https://github.com/KaOSx/core/commit/6b2939d51145fd0ca6660894b5820af5ed3b6e30
